src/binary/non_blocking/raw_binary_reader.rs

```diff
@@ -628,6 +624,12 @@ impl<A: AsRef<[u8]>> StreamReader for RawBinaryBufferReader<A> {
         Box::new(self.annotations_iter())
     }
 
+    fn has_annotations(&self) -> bool {
```

Comment:
🗺️ This method is incidental to this changeset, but is also an important optimization. Readers checking to see if the current value has an annotation can simply see if the encoding had an annotations sequence; this is much faster than the default implementation that constructs a boxed iterator and reports whether it is empty.